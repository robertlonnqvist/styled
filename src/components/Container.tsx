import { JSX, PropsWithChildren } from "react";
import Nav from "./Nav";

const Container = ({
  title,
  headline,
  children,
}: PropsWithChildren<{
  title: string;
  headline: string;
}>): JSX.Element => {
  return (
    <div className="container mx-auto max-w-4xl p-3 md:p-10">
      <Nav title={title} />
      <h1 className="text-2xl text-green-400 font-bold mb-3">{headline}</h1>
      {children}
    </div>
  );
};

export default Container;
