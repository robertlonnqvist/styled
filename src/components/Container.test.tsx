import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import Container from "./Container";

describe("Nav component", () => {
  test("renders Nav component", () => {
    const { container } = render(
      <MemoryRouter>
        <Container title="testTitle" headline="testheadline" />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
  test("changing title on render", () => {
    render(
      <MemoryRouter>
        <Container title="testTitle" headline="testheadline" />
      </MemoryRouter>
    );
    expect(window.document.title).toEqual("Robert LÃ¶nnqvist | testTitle");
  });
});
