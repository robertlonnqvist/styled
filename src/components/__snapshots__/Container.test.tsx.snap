// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Nav component 1`] = `
<div>
  <div
    class="container mx-auto max-w-4xl p-1 md:p-10"
  >
    <header
      class="flex justify-between flex-col md:flex-row items-center"
    >
      <a
        class="font-extrabold text-2xl hover:text-white mb-2"
        href="/"
      >
        <span
          class="text-pink-600 "
        >
          &gt;&gt;
        </span>
         Robert LÃ¶nnqvist
      </a>
      <nav
        class="flex justify-around space-x-2 items-center mb-2"
      >
        <a
          class="font-medium block px-4 py-2 hover:text-white hover:bg-gray-500 rounded-lg "
          href="/"
        >
          About
        </a>
        <a
          class="font-medium block px-4 py-2 hover:text-white hover:bg-gray-500 rounded-lg "
          href="/cv"
        >
          CV
        </a>
        <a
          class="font-medium block px-4 py-2 hover:text-white hover:bg-gray-500 rounded-lg"
          href="https://github.com/robertlonnqvist"
        >
          Code
        </a>
        <a
          class="font-medium block px-4 py-2 hover:text-white hover:bg-gray-500 rounded-lg"
          href="http://www.linkedin.com/in/robertlonnqvist"
        >
          Contact
        </a>
      </nav>
    </header>
    <div
      class="flex flex-col bg-gray-800 rounded-lg shadow-lg"
    >
      <div
        class="flex justify-between items-center px-4 py-2 bg-black rounded-t-lg"
      >
        <div
          class="flex justify-start space-x-2"
        >
          <div
            class="w-3 h-3 bg-red-500 rounded-full"
          />
          <div
            class="w-3 h-3 bg-yellow-400 rounded-full"
          />
          <div
            class="w-3 h-3 bg-green-400 rounded-full"
          />
        </div>
        <div
          class="text-center flex-grow"
        >
          testTitle
        </div>
      </div>
      <div
        class="m-3"
      >
        <h1
          class="text-2xl text-green-400 mb-3"
        >
          testheadline
        </h1>
      </div>
    </div>
  </div>
</div>
`;
